go/consensus/tendermint/apps/roothash/roothash.go

```diff
@@ -476,14 +484,18 @@ func (app *rootHashApplication) FireTimer(ctx *abci.Context, timer *abci.Timer)
 		"timer_round", tCtx.Round,
 	)
 
-	defer state.SetRuntimeState(rtState)
+	defer func() {
+		if err2 := state.SetRuntimeState(ctx, rtState); err2 != nil {
```

Comment:
here too, why do we insist on saving this in case of error? I think whatever calls into this can't recover from anything

Reply:
Good point, this was actually strange even in the previous version.

Reply:
Changed it to not use defer at all.